File: frontend/lib/models/sport_event.dart
class SportEvent {
  final String id;
  final String title;
  final String date;
  final String category;
  final String country;
  final String description;
  final String impactAnalysis;
  final String imageUrl;
  final List<String> relatedEvents;

  SportEvent({
    required this.id,
    required this.title,
    required this.date,
    required this.category,
    required this.country,
    required this.description,
    required this.impactAnalysis,
    required this.imageUrl,
    required this.relatedEvents,
  });

  factory SportEvent.fromJson(Map<String, dynamic> json) {
    return SportEvent(
      id: json['id'] as String,
      title: json['title'] as String,
      date: json['date'] as String,
      category: json['category'] as String,
      country: json['country'] as String,
      description: json['description'] as String,
      impactAnalysis: json['impact_analysis'] as String,
      imageUrl: json['image_url'] as String,
      relatedEvents: List<String>.from(json['related_events'] ?? []),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'date': date,
      'category': category,
      'country': country,
      'description': description,
      'impact_analysis': impactAnalysis,
      'image_url': imageUrl,
      'related_events': relatedEvents,
    };
  }

  int get year => int.parse(date.substring(0, 4));
}

File: app/models/sport_event.py
from pydantic import BaseModel
from typing import List
from datetime import date

class SportEvent(BaseModel):
    id: str
    title: str
    date: str  # Format: YYYY-MM-DD
    category: str  # football, f1, ufc, wrestling, golf, world_cup
    country: str  # UAE, Qatar, Saudi Arabia
    description: str
    impact_analysis: str
    image_url: str
    related_events: List[str]
    
    class Config:
        json_schema_extra = {
            "example": {
                "id": "man-city-takeover-2008",
                "title": "Abu Dhabi Consortium Acquires Manchester City",
                "date": "2008-09-01",
                "category": "football",
                "country": "UAE",
                "description": "Abu Dhabi United Group acquired Manchester City Football Club...",
                "impact_analysis": "Through massive financial investment...",
                "image_url": "https://example.com/man-city-2008.jpg",
                "related_events": ["psg-takeover-2011"]
            }
        }
